app/screens/clock_display/clock_display.android.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+    Platform,
+    TouchableOpacity,
+    Modal,
+    View
+} from 'react-native';
+
+import StatusBar from 'app/components/status_bar';
+import FormattedText from 'app/components/formatted_text';
+import RadioButtonGroup from 'app/components/radio_button';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import ClockDisplayBase from './clock_display_base';
+
+export default class ClockDisplay extends ClockDisplayBase {
+    static propTypes = {
+        showModal: PropTypes.bool.isRequired,
+        close: PropTypes.func.isRequired
+    };
+
+    setMilitaryTime = (value) => {
+        this.setState({
+            militaryTime: value
+        });
+    };
+
+    closeModal = () => {
+        const {militaryTime, close} = this.props;
+        this.setState({
+            militaryTime
+        });
+
+        close();
+    };
+
+    saveSelection = () => {
+        const {militaryTime} = this.state;
+        const {close} = this.props;
+        this.saveClockDisplayPreference(militaryTime);
+        close();
+    };
+
+    renderClockDisplayModal = (style) => {
+        const {showModal} = this.props;
```

Comment:
I'm guessing that if showModal is false you can just return null in the render method and avoid computing everything else, thoughts?

Reply:
react-native Modal component is used to display the Android modal.

showModal is passed to Modal's visible props.  It will be fine the way it is.

FWIW, Modal is the parent component so it won't render anything else